src/Nova/Order.php

```diff
@@ -46,66 +51,44 @@ public static function indexQuery(NovaRequest $request, $query)
 
     public function fieldsForIndex(NovaRequest $request)
     {
-        return [
+        return array_filter([
             ID::make()->sortable(),
             Text::make('Order Number')->sortable(),
-            BelongsTo::make('Customer', 'customer', app()->getAlias('nova.customer'))->sortable(),
-            BelongsTo::make('Location', 'location', app()->getAlias('nova.location'))->sortable(),
+            nova('customer') ? BelongsTo::make('Customer', 'customer', nova('customer'))->sortable() : null,
+            nova('location') ? BelongsTo::make('Location', 'location', nova('location'))->sortable() : null,
```

Comment:
This pattern is used in both `fieldsForIndex(...)` and `fields(...)`.  Those relationships that might not exist (basically, any where `nova(..)` helper is required) are modified to only include the relationship if the nova resource exists (indicated by non-null response from the `nova(...)` helper).  Because this might result in null values, the resulting array is passed through array_filter to remove them.

This approach allows the optional fields/relations to appear in the desired order w.r.t. other fields being displayed.

Reply:
Would this prevent the `->required()` and even the `->nullable()` methods on the nova fields?

Reply:
The entire expression -- with any modifiers -- is either fully evaluated or skipped entirely.    The only difference is the optional dependencies have `nova('...') ?` prepended before the existing expression and ` : null,` appended after it (with the expression itself being modified to use the `nova(...)` helper for the resource class).

Reply:
Great!